### PR TITLE
fixed issue where you can't select secondary handles (in/out) for curve 3d in top view mode

### LINKED ISSUE #85795: 4.2 - Path3D Control Nodes - Cannot create bezier handles in top down ortho view.
### Tested versions

works correctly in v4.1.3.stable.official [f06b6836a] as it doesnt have the Tilt Handles in editor.

issue occurs in v4.2.stable.official [46dc27791] as tilt handles were introduced.

### System information

Godot v4.2.stable - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3050 Laptop GPU (NVIDIA; 31.0.15.4592) - AMD Ryzen 5 4600H with Radeon Graphics (12 Threads)

### Issue description

with the path3d node and looking in top down orthgraphic view cannot use control point button to drag and create bezier handles.
only the tilt handle is selectable due to it being higher up and making the control point unclickable to drag.

### Steps to reproduce

all i done was in 4.2 was create a new 3d scene.
create a path3d node.
into top down ortho view.
created a few points with the image button.

then, using the select control points button 
image
to try and create bezier handles.

this only indicates that the tilt handle has been selected and cannot create beziers.

my only workaround was to look from bottom up ortho, create the bezier handles and then return to top down ortho view to adjust placements.

possible solution would maybe have another button that hides or only controls the tilt handles

image

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by jejopl**
The issue: you can't modify 3d curve using secondary handles in top view mode. It's because there's no way to click on handle in/out because handle tilt is on top of them and you are always clicking on handle tilt

https://github.com/godotengine/godot/assets/32012330/cb2e1d97-374b-41b6-95fb-8c423705b117

The solution:
When you are in top view, and you click on handle tilt, treat it like you have clicked handle in or handle out for the beginning of the curve

https://github.com/godotengine/godot/assets/32012330/d9e405ff-c6f6-4d4e-911b-87daecf34d3f

* *Bugsquad edit, fixes: https://github.com/godotengine/godot/issues/85795*

### CODE REVIEW

File: editor/plugins/path_3d_editor_plugin.cpp [hunk 1]
```diff
@@ -41,6 +41,9 @@
 #include "scene/gui/menu_button.h"
 #include "scene/resources/curve.h"
 
+// A flag to indicate if the camera is top view.
+bool is_top_view = false;
```
Comment (AThousandShips) [inline editor/plugins/path_3d_editor_plugin.cpp:45]: This shouldn't be just a global boolean, it should be in the header

**Review by AThousandShips — DISMISSED** [review]
Thank you for your contribution! And welcome!

If there isn't already a bug report for this issue please open one to make this easier to track :)

File: editor/plugins/path_3d_editor_plugin.cpp (hunk 1 above)
Comment (jejopl) [reply editor/plugins/path_3d_editor_plugin.cpp:45]: @AThousandShips thanks for the feedback! This is my first ever contribution to c++ code, I'm working in JavaScript on daily basis 😅

Comment (AThousandShips) [reply editor/plugins/path_3d_editor_plugin.cpp:45]: You forgot to remove this line :)

**Comment by jejopl** [comment]
@AThousandShips the issue has been already created **https://github.com/godotengine/godot/issues/85795**

**Comment by AThousandShips** [comment]
(No need to tag me, it just adds unnecessary pings :))

**Comment by GreenCrowDev** [comment]
Isn't this too much of a workaround? I think the viewport editor UI is just too cluttered, we should design better curve management instead of very specific workaround imho.
This PR tackles the same issue I tried to tacke with this https://github.com/godotengine/godot-proposals/issues/8890
I would suggest to open a new proposal issue to discuss together how to improve general usage in an elegant way.

**Comment by jejopl** [comment]
Yeah, you're right

**Comment by akien-mga** [comment]
Superseded by #89847. Thanks for the contribution!

**Pull request closed without merging**